challenges/2021-06-01-find-three-largest-numbers/Solution.java

```diff
@@ -20,10 +20,44 @@
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import java.util.Arrays;
+
 public class Solution {
     static int[] findThreeLargestNumbers(int[] array) {
         // Insert your solution here
-        return new int[]{0, 0, 0};
+        // the quickest answer is
+        // Arrays.sort(array);
+        // int[] threeAmigos = int[]{array[arrayLength - 1], array[arrayLength -2], ..}
+        int lengthArray = array.length;
+        if (lengthArray < 3) {
+            return null;
+        }
+
+        int min,mid,max;
+        min = array[0];
+        mid = array[1];
+        max = array[2];
+        for (int i = 0; i < lengthArray; i++) {
+            if (array[i] > mid) {
```

Comment:
I  think you can combine the code of this nested if and nested else something like, 
if cond {
} else if {
} else if {
}

It is not necessary to always use a else part. I think that will simplify the code.

Reply:
I agree, I think this could be simplified to something like:
```
if (array[i] > min && array[i] <= mid) {
    min = array[i];
} else if array[i] > mid && array[i] <= max {
    min = mid;
    mid = array[i];
} else if array[i] > max {
    min = mid;
    mid = max;
    max = array[i];
}
```

I'm not sure if my binary operator syntax is correct here, but you get the picture! Less LOC and IMO more clarity.